interfaces.go

```diff
@@ -79,6 +79,9 @@ type Method interface {
 
 	// Stores the next location to evaluate in xNext
 	Iterate(loc *Location, xNext []float64) (EvaluationType, IterationType, error)
+
+	// Requires returns the minimum requirements of Method on Function.
```

Comment:
My intention for `Requires()` was to just provide information what kind of optimizer the Method is (derivative-free, gradient-based, ...), without considering how this information is used and how it is translated into interfaces that Function should satisfy. That's why I kept the comment rather minimal, because `Requires()` is of interest only to implementers of external minimizers and to us, not to users of optimize.

On the other hand, matching the requirements of Method and capabilites of Function is tricky and should be clearly explained in the documentation because it directly affects the users of optimize, it tells them what interfaces Function should satisfy. But `Requires()` is not the right place for such explanation, I would say that `Local()` or the package docs would be a better place. I think that evaluations returned by `Init()` and `Iterate()` are more important for the matching between requirements and the methods of Function, maybe each optimizer should indicate the evaluations it requests in its docs.

I will try to improve the comment for `Requires()` anyway.